### PR TITLE
[TypeScript] Add orderbook JSON types

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds orderbook JSON types

This change was included as part of #736 but I think it deserves some special attention.

### Test Plan

Unit tests validate serialization and deserialization.

### CODE REVIEW

File: src/orderbook.ts [hunk 1]
```diff
@@ -293,6 +301,14 @@ export class Orderbook {
   }
 }
 
+export interface OrderbookJson {
+  baseToken: string;
+  quoteToken: string;
+  remainingFractionAfterFee: Fraction | FractionJson;
+  asks: Record<string, Offer | OfferJson>;
+  bids: Record<string, Offer | OfferJson>;
```
Comment (e00E) [inline src/orderbook.ts]: Why allow this flexibility? If it comes from json then it should always be `OfferJson`, right?

**Review by e00E — APPROVED** [review]

File: src/orderbook.ts (hunk 1 above)
Comment (nlordell) [reply src/orderbook.ts]: Its because `toJSON` method doesn't convert `Offer` to `OfferJson`.

Comment (nlordell) [reply src/orderbook.ts]: Hmm... I guess I can create 2 types here instead of just one. That might be a more correct approach.

Comment (nlordell) [reply src/orderbook.ts]: @e00E I split it into 2 types now. Let me know if you think this is better.

**Pull request merged**